dask/dataframe/core.py

```diff
@@ -629,12 +647,23 @@ def _loc(df, start, stop, include_right_boundary=True):
     2  20
     2  30
     """
-    result = df.loc[slice(start, stop)]
+    result = df.loc[start:stop]
     if not include_right_boundary:
         # result = df[df.index != stop]
         result = result.iloc[:result.index.get_slice_bound(stop, 'left',
                                                    result.index.inferred_type)]
-    return result
+    if not result.empty:
+        return result
+
+    # this is horrible
+    # i need to find a non-type-specific way to generate a range containing
+    # values that don't exist in the index and fill that with NaNs
+    d = np.diff(df.index.values)
```

Comment:
``pd.infer_freq`` might help here as it looks at the index directly (only works for datetimelike though), but you can catch the ``TypeError`` and do something else (e.g. its an ``Int64Index`` at this point)